library/Opus/Search/Result/Match.php

```diff
@@ -0,0 +1,265 @@
+		}
+
+		return $this->doc;
+	}
+
+	/**
+	 * Assigns score of match in context of related search.
+	 *
+	 * @param $score
+	 * @return $this
+	 */
+	public function setScore( $score ) {
+		if ( !is_null( $this->score ) ) {
+			throw new RuntimeException( 'score has been set before' );
+		}
+
+		$this->score = floatval( $score );
+
+		return $this;
+	}
+
+	/**
+	 * Retrieves score of match in context of related search.
+	 *
+	 * @return float|null null if score was not set
+	 */
+	public function getScore() {
+		return $this->score;
+	}
+
+	/**
+	 * Retrieves matching document's number in series selected by its ID.
+	 *
+	 * This method is provided for downward compatibility. You are advised to
+	 * inspect document's model for this locally available information rather
+	 * than relying on search engine returning it.
+	 *
+	 * @deprecated
+	 * @return string
+	 */
+	public function getSeriesNumber( $seriesId ) {
+		if ( !$seriesId ) {
+			return null;
+		}
+
+		if ( !is_array( $this->seriesNumbers ) ) {
+			$this->seriesNumbers = array();
+
+			foreach ( $this->getDocument()->getSeries() as $linkedSeries ) {
+				$id     = $linkedSeries->getModel()->getId();
+				$number = $linkedSeries->getNumber();
+
+				$this->seriesNumbers[$id] = $number;
+			}
+		}
+
+		return array_key_exists( $seriesId, $this->seriesNumbers ) ? $this->seriesNumbers[$seriesId] : null;
+	}
+
+	/**
+	 * Assigns timestamp of last modification to document as tracked in search
+	 * index.
+	 *
+	 * @note This information is temporarily overloading related timestamp in
+	 *       local document.
+	 *
+	 * @param {int} $timestamp Unix timestamp of last modification tracked in search index
+	 * @return $this fluent interface
+	 */
+	public function setServerDateModified( $timestamp ) {
+		if ( !is_null( $this->serverDateModified ) ) {
+			throw new RuntimeException( 'timestamp of modification has been set before' );
+		}
+
+		$this->serverDateModified = new Opus_Date();
+
+		if ( ctype_digit( $timestamp = trim( $timestamp ) ) ) {
+			$this->serverDateModified->setUnixTimestamp( intval( $timestamp ) );
+		} else {
+			$this->serverDateModified->setFromString( $timestamp );
+		}
+
+		return $this;
+	}
+
+	/**
+	 * Provides timestamp of last modification preferring value provided by
+	 * search engine over value stored locally in document.
+	 *
+	 * @note This method is used by Opus to detect outdated records in search
+	 *       index.
+	 *
+	 * @return Opus_Date
+	 */
+	public function getServerDateModified() {
+		if ( !is_null( $this->serverDateModified ) ) {
+			return $this->serverDateModified;
+		}
+
+		return $this->getDocument()->getServerDateModified();
+	}
+
+	/**
+	 * Passes all unknown method invocations to related instance of
+	 * Opus_Document.
+	 *
+	 * @param string $method name of locally missing/protected method
+	 * @param mixed[] $args arguments used on invoking that method
+	 * @return mixed
+	 */
+	public function __call( $method, $args ) {
```

Comment:
Method `Remotecontrol_Model_DownloadList::prepareCsv` calls `getTitle`. However there is no field "Title" in Opus_Document. In the old class, `Opus_SolrSearch_Result` the value was returned as part of the response from Solr and I believe for many requests no instance of the actual document was created. I think that was at least partly because of performance considerations. Replacing getTitle with getTitleMain will also not be enough, since there can be multiple main titles in different languages. That maybe a problem with the data model, however we cannot tackle all issues at once.